src/unix/core.c

```diff
@@ -983,3 +987,47 @@ int uv__dup2_cloexec(int oldfd, int newfd) {
     return r;
   }
 }
+
+
+int uv_homedir(char* home, size_t* len) {
+
+  char login[LOGIN_NAME_MAX];
+  struct passwd pw;
+  struct passwd* result;
+  char* buf;
+  size_t loginsize = sizeof(login);
+  size_t bufsize = sysconf(_SC_GETPW_R_SIZE_MAX);
```

Comment:
Minor style issue: the code in src/unix does not normally use the C99 declare + assign syntax.